.github/workflows/test_cli_cuda_pytorch_single_gpu.yaml

```diff
@@ -49,4 +49,5 @@ jobs:
             --workdir /workspace
           run: |
             pip install -e .[testing,diffusers,timm,peft,bitsandbytes,autoawq,auto-gptq]
-            pytest -x -s -k "cli and cuda and pytorch and not (dp or ddp or device_map or deepspeed) and not (awq)"
+            python scripts/install_quantization_libs.py --install-autoawq-from-source --install-autogptq-from-source
+            pytest -x -s -k "cli and cuda and pytorch and not (dp or ddp or device_map or deepspeed) and not (awq)" --ignore=external_repos
```

Comment:
Good point. The problem is that the kernels are built for a specific GPU architecture. So, if we build them on an A100, they might not run on a T4 because the targeted architectures would be different. One option could be to create separate Docker images for each architecture, but this seems overly complicated

Reply:
it doesn't have to be built for one architecture, you can use this var env to control the targeted archs https://github.com/huggingface/optimum-benchmark/blob/main/docker/cuda-ort/Dockerfile#L42

Reply:
it will be python version (cpython) and os dependent but that's not an issue here since we use the same python version.

Reply:
Thanks for the tip. Indeed `TORCH_CUDA_ARCH_LIST` fixed the issue. How did you find out which archs to use? I used the one from https://github.com/AutoGPTQ/AutoGPTQ/blob/main/.github/workflows/build_wheels_cuda_linux.yml which is `TORCH_CUDA_ARCH_LIST="6.0 6.1 7.0 7.5 8.0 8.6 8.9 9.0+PTX"` but I am unsure which archs to target

Reply:
i used the list from https://pytorch.org/docs/stable/generated/torch.cuda.get_arch_list.html
which has every arch that torch (specific version) was compiled for.

Reply:
Thanks, I update the list to have same TORCH_CUDA_ARCH_LIST as torch 2.4.0